src/parser/paladin/protection/modules/spells/LightOfTheProtector.js

```diff
@@ -0,0 +1,158 @@
+  _delays = [];
+
+  _activeSpell = SPELLS.LIGHT_OF_THE_PROTECTOR;
+
+  _overhealing = 0;
+  _actualHealing = 0;
+
+  constructor(props) {
+    super(props);
+    if(this.selectedCombatant.hasTalent(SPELLS.HAND_OF_THE_PROTECTOR_TALENT.id)) {
+      this._activeSpell = SPELLS.HAND_OF_THE_PROTECTOR_TALENT;
+    }
+
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this._startDelayTimer);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(this._activeSpell), this._countDelay);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(this._activeSpell), this._countHeal);
+
+    if(this.selectedCombatant.hasTalent(SPELLS.RIGHTEOUS_PROTECTOR_TALENT.id)) {
+      this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SHIELD_OF_THE_RIGHTEOUS), this._updateDelayRP);
+    }
+  }
+
+  _startDelayTimer(event) {
+    this._lastHit = event;
+    this._msTilHeal = this.spells.isAvailable(this._activeSpell.id) ? 0 : this.spells.cooldownRemaining(this._activeSpell.id);
+  }
+
+  // update the delay based on SotR cast with the RP talent, which
+  // reduces LotP/HotP CD by 3s
+  _updateDelayRP(event) {
+    const delayFromHit = event.timestamp - this._lastHit.timestamp;
+    this._msTilHeal -= RP_REDUCTION_TIME;
+    // we couldn't cast the heal before the current event happened
+    if(this._msTilHeal < delayFromHit) {
+      this._msTilHeal = delayFromHit;
+    }
+  }
+
+  _countDelay(event) {
+    const delay = event.timestamp - this._lastHit.timestamp - this._msTilHeal;
+    if(delay < 0) {
+      console.error("LotP/HotP delay came out negative", delay);
+    }
+    this._delays.push(delay);
+
+    if(delay < HEAL_DELAY_THRESHOLD) {
+      return; // nothing left to do
+    }
+
+    const meta = event.meta || {
+      inefficientCastReason: `This ${this._activeSpell.name} cast was inefficient because:`,
+    };
+    meta.inefficientCastReason += `<br/> - You delayed casting it for <b>${(delay / 1000).toFixed(2)}s</b> after being hit.`;
+    meta.isInefficientCast = true;
+    event.meta = meta;
+  }
+
+  _countHeal(event) {
+    const overhealing = event.overheal || 0;
+    this._overhealing += overhealing;
+    this._actualHealing += event.amount;
+
+    if(overhealing === 0 || event.hitType === HIT_TYPES.CRIT) {
+      return; // not gonna penalize overhealing on crits because it heals for SO MUCH
+    }
+
+    const meta = event.meta || {
+      inefficientCastReason: `This ${this._activeSpell.name} cast was inefficient because:`,
+    };
+    meta.inefficientCastReason += `<br/> - You cast it while at high health, causing it to overheal for ${formatNumber(overhealing)}.`;
+    meta.isInefficientCast = true;
+    event.meta = meta;
+  }
+
+  get avgDelay() {
+    if(this._delays.length === 0) {
+      return 0;
+    }
+
+    return this._delays.reduce((sum, delay) => sum + delay, 0) / this._delays.length;
+  }
+
+  get overhealRatio() {
+    return this._overhealing / (this._overhealing + this._actualHealing);
```

Comment:
I know it's unrealistic in real logs, but just in the inifitely small chance that there were no heals with the spell, can you check for division by zero?
```suggestion
    return (this._overhealing / (this._overhealing + this._actualHealing)) || 0;
```